package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -265,7 +265,8 @@ cdef class DCDFile:
             return
 
         current_frame = state[1]
-        self.seek(current_frame)
+        self.seek(current_frame - 1)
```

Comment:
Do these lines make the low-level tests fail?

Reply:
yeah...it messed up with the position of the file.

The fix is ditching current changes, and changing line 410 to:
```python
        if frame >= self.n_frames + 1:
```
Besides, I don't think current tests cover for the "true" last frame i.e.:
```python
        traj = DCDReader(DCD)
        traj[-1]  #  move to last frame
        dcd = traj._file
        dcd.tell() == len(dcd)  #  instead of len(dcd) - 1
```